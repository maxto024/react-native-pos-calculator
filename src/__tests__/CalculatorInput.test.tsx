import { shallow } from 'enzyme'
import * as React from 'react'
import { NativeModules } from 'react-native'
import { CalculatorInput } from '../CalculatorInput'

NativeModules.UIManager = {
  measureLayoutRelativeToParent: jest.fn()
}

it('renders calculator input correctly', () => {
  const wrapper = shallow(<CalculatorInput />)
  expect(wrapper).toMatchSnapshot()
})
