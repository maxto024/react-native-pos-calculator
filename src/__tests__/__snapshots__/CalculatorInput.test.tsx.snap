// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calculator input correctly 1`] = `
<Component
  onLayout={[Function]}
>
  <Component
    style={
      Array [
        Object {
          "borderBottomColor": "grey",
          "borderBottomWidth": 1,
          "height": 24,
          "margin": 10,
        },
        undefined,
        Object {},
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 16,
            },
            undefined,
            Object {},
          ]
        }
      >
        0
      </Text>
    </TouchableOpacity>
  </Component>
  <Component
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.3)",
              "flex": 1,
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <Calculator
          acceptButtonBackgroundColor="#14CC60"
          acceptButtonColor="#ffffff"
          actionButtonBackgroundColor="#e7e5e3"
          actionButtonColor="#000000"
          borderColor="#52525B"
          calcButtonBackgroundColor="#ff8d00"
          calcButtonColor="#ffffff"
          decimalSeparator="."
          displayBackgroundColor="#ffffff"
          displayColor="#000000"
          displayTextAlign="left"
          fontSize={18}
          hasAcceptButton={true}
          height={889.3333333333333}
          numericButtonBackgroundColor="#ffffff"
          numericButtonColor="#aaaaaa"
          onAccept={[Function]}
          prefix=""
          suffix=""
          thousandSeparator=","
          value={0}
          width={750}
        />
      </Component>
    </TouchableWithoutFeedback>
  </Component>
</Component>
`;
